### DISCUSSION TITLE
Test with intercept succeeds locally but fails on CircleCi

### DISCUSSION THREAD

**Discussion by katia-aa** [cypress-io/cypress#9597] [Questions and Help]
I am trying to use `cy.intercept` to stub an api. **The problem occurs on CircleCi**. Locally the test runs fine. 

I am using cypress 6.1.0

Here is my test:

```javascript
describe('Ask', () => {
  it('search for a furnace', () => {
    cy.intercept('GET', '/v1/answers', {
      fixture: 'answers.json',
    }).as('getAnswers')

    cy.visit('/ask')
    cy.get('input[id="searchQuery"]').should('be.visible')
    cy.get('input[id="searchQuery"]').type('furnace')
    cy.wait(5000)
    cy.wait('@getAnswers')
    .its('response')
    .should('deep.include', {
      statusCode: 200,
    })
    .and('have.property', 'body')
    .then(body => {
      expect(body).to.be.an('object')
    })
})
```

The error I am getting on circle ci looks like this:

<img width="1311" alt="Screen Shot 2020-12-09 at 1 43 58 PM" src="https://user-images.githubusercontent.com/11549223/101677645-d74b3600-3a2a-11eb-8c48-e02300fe540c.png">

I was able to access the screenshot saved by CircleCi and it looks like this 

![Ask -- search for a furnace (failed)](https://user-images.githubusercontent.com/11549223/101678018-5a6c8c00-3a2b-11eb-983a-ba38e06198ac.png)

From the debugging I've done, it seems like the request never actually completes on CircleCi. I tried the following which didn't work: 

- chaining `wait('@getAnswers').then(() => /* assert things */))`
- Increasing timeout `wait('@getAnswers', {timeout: 60000})` and up to 2 minutes 
- Adding an additional `wait(20000)` after typing 'furnace' in the test

none of these are working on CircleCi and all produce the same error: `No request ever occurred`. 

I thought maybe I'd find my answer here: https://glebbahmutov.com/blog/cypress-intercept-problems/
But none of the gotchas seem to relate to my issue. 

In case there was a caching issue on CircleCi - I did try deleting `if-none-match`: 

```javascript
cy.intercept('GET', '/v1/answers', req => {
      delete req.headers['if-none-match']

      req.reply((res) => {
        // sends a fixture body instead of the existing 'res.body'
        res.send({ fixture: 'answers.json' })
      })
    }).as('getAnswers')
```

but that wasn't it either.

**Comment by bahmutov** [comment]
That is very weird, when you run locally it shows the "getAnswers" badge on that XHR to `/v1/answers`?

**Reply by katia-aa** [reply]
It does show the `getAnswers`  badge like this:

<img width="858" alt="Screen Shot 2020-12-09 at 2 53 20 PM" src="https://user-images.githubusercontent.com/11549223/101680297-79205200-3a2e-11eb-8de1-acd4f0508561.png">

**Comment by bahmutov** [comment]
It seems to me that `GET /v1/answers` never comes back with the response, thus it has no status code, just 200. Why it would not work, it is hard to say. But also, I don't see `(req)` with `getAnswers` badge for that when running on CI

**Reply by katia-aa** [reply]
I think it might be the corporate proxy - I was able to reproduce the error locally by turning off my VPN

**Reply by katia-aa** [reply]
@bahmutov I am a little bit confused about intercept. I thought the idea was that if I provide a fixture the request will NOT go to the server as stated in docs:

> If cy.intercept provides a response object, or a fixture, or calls req.reply() then the request will NOT go to the server, and instead will be mocked from the test.

but it seems like it does try to go to the server...

**Reply by briansmith-fr** [reply]
I'm curious whether you have made progress on this; I think I am seeing something similar: https://github.com/cypress-io/cypress/discussions/9578

It's very frustrating that it works locally, but fails in github-actions.

**Reply by bahmutov** [reply]
Can I ask if this request is CORS? Can you open the DevTools Network panel locally, turn off VPN and see if there is a failing OPTIONS call before the actual call?
